bsmetadata/metadata_utils.py

```diff
@@ -183,14 +183,33 @@ def _collate_metadata(metadata_list: List[dict], cfg: MetadataConfig):
         assert metadata_node.relative_start_pos not in metadata_dict_idx[metadata_node.char_start_idx]
         assert metadata_node.relative_end_pos not in metadata_dict_idx[metadata_node.char_end_idx]
 
-        metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
-        metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
+        if start_text:
+            metadata_dict_idx[metadata_node.char_start_idx][metadata_node.relative_start_pos] = start_text
+        if end_text:
+            metadata_dict_idx[metadata_node.char_end_idx][metadata_node.relative_end_pos] = end_text
```

Comment:
Why do we need these `if start_text:` and `if end_text:` checks here?

Reply:
Let me answer this question as I was the one who advised @manandey  to make this change (and we discussed it outside github - not a good practice indeed :disappointed: ).

These checks allow to use the same dataset to do experiments in `beg` mode and in `end` mode. So we don't have to rework the dataset upstream to do one or the other experiment